ls1intum/Artemis
# [Enhancement] Java template and SCA configuration

### Checklist

### Motivation and Context


Improves the default static code analysis configuration and provides examples for instructors how to configure the tools. Fixes static code analysis findings in the templates.

### Description


- Add config xml files for all tools
- On exercise creation, copy the configs to the test repository if static code analysis is enabled
- Fix issues in templates

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create programming exercises with static code analysis (one with eclipse and one with maven template)
4. Check the test repository
--> SCA XML configs should be copied to the staticCodeAnalysisConfig directory
5. Check the template and solution repository
--> Template: 0 Issues, Solution: 0 Issues
6. Create a programming exercise without SCA
7. Check the test repository
--> the directory staticCodeAnalysisConfig should not exist

### Test Coverage

Nothing should have changed
